pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Hook.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.api;
+
+import org.apache.pulsar.common.classification.InterfaceAudience;
+
+/**
+ * An interface for hook
+ * Initial and close of external resources
+ */
+@InterfaceAudience.Public
+public interface Hook {
```

Comment:
I suggest this new api extends the `Function` interface since it's strongly coupled with the `Function` api and should not be used individually in other cases.

```
public interface HookFunction extends Function {
    
    void setup();

    void cleanup();
}
```

And we should keep `java.util.Function` as it is to keep its simplicity.

Reply:
I have modified it, please check